diff --git a/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Interfaces/ITransactieRepository.cs b/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Interfaces/ITransactieRepository.cs
index 98f7eb9..dddb2f9 100644
--- a/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Interfaces/ITransactieRepository.cs
+++ b/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Interfaces/ITransactieRepository.cs
@@ -1,4 +1,5 @@
-﻿using System;
+﻿using ScradaCadeauBonnen_BL.Models;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
@@ -6,7 +7,8 @@ using System.Threading.Tasks;
 
 namespace ScradaCadeauBonnen_BL.Interfaces
 {
-    interface ITransactieRepository
+    public interface ITransactieRepository
     {
+        void BewaarTransactie(Transactie transactie);
     }
 }
diff --git a/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Manager/TransactieBeheerder.cs b/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Manager/TransactieBeheerder.cs
index 775bd20..6932cad 100644
--- a/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Manager/TransactieBeheerder.cs
+++ b/Scrada_CadeauBonnen/ScradaCadeauBonnen_BL/Manager/TransactieBeheerder.cs
@@ -1,4 +1,6 @@
-﻿using System;
+﻿using ScradaCadeauBonnen_BL.Interfaces;
+using ScradaCadeauBonnen_BL.Models;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
@@ -6,7 +8,18 @@ using System.Threading.Tasks;
 
 namespace ScradaCadeauBonnen_BL.Manager
 {
-    internal class TransactieBeheerder
+    public class TransactieBeheerder
     {
+        private ITransactieRepository repo;
+
+        public TransactieBeheerder(ITransactieRepository repo)
+        {
+            this.repo = repo;
+        }
+
+        public void BewaarTransactie(Transactie transactie)
+        {
+            repo.BewaarTransactie(transactie);
+        }
     }
 }
diff --git a/Scrada_CadeauBonnen/ScradaCadeauBonnen_DL/Repo/TransactieRepository.cs b/Scrada_CadeauBonnen/ScradaCadeauBonnen_DL/Repo/TransactieRepository.cs
index 843fa53..9770a92 100644
--- a/Scrada_CadeauBonnen/ScradaCadeauBonnen_DL/Repo/TransactieRepository.cs
+++ b/Scrada_CadeauBonnen/ScradaCadeauBonnen_DL/Repo/TransactieRepository.cs
@@ -1,4 +1,7 @@
-﻿using System;
+﻿using Microsoft.Data.SqlClient;
+using ScradaCadeauBonnen_BL.Interfaces;
+using ScradaCadeauBonnen_BL.Models;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
@@ -6,8 +9,32 @@ using System.Threading.Tasks;
 
 namespace ScradaCadeauBonnen_DL.Repo
 {
-    public class TransactieRepository
+    public class TransactieRepository : ITransactieRepository
     {
-        //testing
+        private string connectionString;
+
+        public TransactieRepository(string connectionString)
+        {
+            this.connectionString = connectionString;
+        }
+
+        public void BewaarTransactie(Transactie transactie)
+        {
+            using (SqlConnection conn = new SqlConnection(connectionString)) 
+            using (SqlCommand cmd = conn.CreateCommand())
+            {
+                string query = @"INSERT INTO Transactie (CadeaubonId, Bedrag, Datum, Reden, Status)
+                         VALUES (@CadeaubonId, @Bedrag, @Datum, @Reden, @Status)";
+
+                cmd.Parameters.AddWithValue("@CadeaubonId", transactie.CadeaubonId);
+                cmd.Parameters.AddWithValue("@Bedrag", transactie.Bedrag);
+                cmd.Parameters.AddWithValue("@Datum", transactie.Datum);
+                cmd.Parameters.AddWithValue("@Reden", transactie.Reden);
+                cmd.Parameters.AddWithValue("@Status", transactie.Status);
+
+                conn.Open();
+                cmd.ExecuteNonQuery();
+            }
+        }
     }
 }
diff --git a/Scrada_CadeauBonnen/ScradaCadeauBonnen_WPF/Bevestiging.xaml.cs b/Scrada_CadeauBonnen/ScradaCadeauBonnen_WPF/Bevestiging.xaml.cs
index 5f99ec1..af310f2 100644
--- a/Scrada_CadeauBonnen/ScradaCadeauBonnen_WPF/Bevestiging.xaml.cs
+++ b/Scrada_CadeauBonnen/ScradaCadeauBonnen_WPF/Bevestiging.xaml.cs
@@ -30,6 +30,7 @@ namespace ScradaCadeauBonnen_WPF
             InitializeComponent();
             BestellingItems.Add("Cadeaubon €25");
             BestellingItems.Add("Cadeaubon €10");
+            BestellingItems.Add("Cadeaubon €1500");
             InfoBestelling.ItemsSource = BestellingItems;
         }
 
@@ -42,7 +43,7 @@ namespace ScradaCadeauBonnen_WPF
                 var httpClient = new HttpClient();
                 var request = new PaymentRequest
                 {
-                    Amount = 2500,
+                    Amount = totaalBedragInCenten,
                     Currency = "eur"
                 };
 
@@ -56,6 +57,7 @@ namespace ScradaCadeauBonnen_WPF
                         // Open Stripe-url in de standaardbrowser inplaats van een nieuwe window want dit gaf teveel fouten en problemen
                         Process.Start(new ProcessStartInfo
                         {
+                            //hier komt de url van de api and word geset naar FileName
                             FileName = result.Url,
                             UseShellExecute = true
                         });
